### PR TITLE
Fix importing by ISIN number

### PR DESCRIPTION
**Pull request by sahb1239**
When importing using CSV or JSON by ISIN number it the Yahoo finance provider will fail to get historical data due to the the ISIN number is not mapped to the symbol used by Yahoo Finance.

This sample CSV illustrates the problem:
```
Date,Account,Action,DataSource,Code,Quantity,unitPrice,Fee,Currency
2024-02-25,My Account,buy,YAHOO,DK0060442556,5,275.10,0,DKK
2024-02-25,My Account,buy,YAHOO,SPIC25KL.CO,5,275.10,0,DKK
```

Without the fix this will map to 2 different symbols as seen on the import screen here:
Screenshot 2024-02-25 154439

This leads to that the historical data gathering is failing and the same stock being shown twice:
image

This fix ensures that the correct symbol is returned from the yahoo-finance-service.ts. When trying to import the same sample file it now maps to the same symbol instead.
Screenshot 2024-02-25 154340

### CODE REVIEW

**Review by dtslvr — CHANGES_REQUESTED** [review]
The idea is great, but it breaks some other functionality: when you import `BTCUSD`, it shows as `BTC-USD` instead of `BTCUSD` (see `ok-btc.csv`).
I think this is because `yahooFinanceDataEnhancerService.getAssetProfile(symbol)`expects a symbol converted to _Yahoo_ format (via `convertToYahooFinanceSymbol()`).

**Comment by sahb1239** [comment]
Nice catch - will take a look at this again later today / this week.

**Comment by sahb1239** [comment]
> The idea is great, but it breaks some other functionality: when you import `BTCUSD`, it shows as `BTC-USD` instead of `BTCUSD` (see `ok-btc.csv`). I think this is because `yahooFinanceDataEnhancerService.getAssetProfile(symbol)`expects a symbol converted to _Yahoo_ format (via `convertToYahooFinanceSymbol()`).

Seems like the root cause is that yahoo-finance.service.ts -> search automatically removes '-' from the symbol. Using the import function it's possible to import with both symbol formats:
image

Both formats are currently valid due to the convertToYahooFinanceSymbol method and this also means that these symbols right now might be tracked multiple times and might require data migration if you want to combine these symbols.

I think the question now is which behavior you prefer - BTCUSD or BTC-USD which would mean either changing the search functionality or the import functionality. For the potential data migration I think that's best left in a separate issue where consequences / approaches can be discussed.

**Comment by sahb1239** [comment]
I added the potential import fix where the format BTCUSD will be saved.

File: apps/api/src/services/data-provider/data-enhancer/yahoo-finance/yahoo-finance.service.ts [hunk 1]
```diff
@@ -228,6 +228,11 @@ export class YahooFinanceDataEnhancerService implements DataEnhancerInterface {
             { name: assetProfile.summaryProfile?.sector, weight: 1 }
           ];
         }
+      } else if (assetSubClass == AssetSubClass.CRYPTOCURRENCY) {
```
Comment (dtslvr) [inline apps/api/src/services/data-provider/data-enhancer/yahoo-finance/yahoo-finance.service.ts:231]: Can you remove this block and try to replace the following line above

```
response.symbol = assetProfile.price.symbol;
```

with

```
response.symbol = this.convertFromYahooFinanceSymbol(assetProfile.price.symbol);
```

File: apps/api/src/services/data-provider/yahoo-finance/yahoo-finance.service.ts [hunk 2]
```diff
@@ -38,17 +38,10 @@ export class YahooFinanceService implements DataProviderInterface {
   }: {
     symbol: string;
   }): Promise<Partial<SymbolProfile>> {
-    const { assetClass, assetSubClass, currency, name } =
+    const response =
```
Comment (dtslvr) [inline apps/api/src/services/data-provider/yahoo-finance/yahoo-finance.service.ts:41]: This is simpler 🙂

```
return this.yahooFinanceDataEnhancerService.getAssetProfile(symbol);
```

**Review by dtslvr — CHANGES_REQUESTED** [review]

**Comment by dtslvr** [comment]
> I think the question now is which behavior you prefer - BTCUSD or BTC-USD which would mean either changing the search functionality or the import functionality.

`BTCUSD`

File: apps/api/src/services/data-provider/data-enhancer/yahoo-finance/yahoo-finance.service.ts (hunk 1 above)
Comment (sahb1239) [reply apps/api/src/services/data-provider/data-enhancer/yahoo-finance/yahoo-finance.service.ts]: Solved

File: apps/api/src/services/data-provider/yahoo-finance/yahoo-finance.service.ts (hunk 2 above)
Comment (sahb1239) [reply apps/api/src/services/data-provider/yahoo-finance/yahoo-finance.service.ts]: Solved

**Comment by sahb1239** [comment]
I updated the PR and retested with the sample files:
image

**Review by dtslvr — APPROVED** [review]
Thank you @sahb1239!

I have made some minor improvements (https://github.com/ghostfolio/ghostfolio/pull/3051/commits/8b08a1bf2ba22a573f8af1fbc3ee76020a49431e), ready to merge now.

**Pull request merged**